core/common/src/MapImagerySettings.ts

```diff
@@ -6,47 +6,221 @@
+        mapType:this.mapType,
+        visible:this.visible,
+        transparency:this.transparency,
+      },
+    };
+  }
+  private mapTypeName() {   // TBD.. Localization.
+    switch (this.mapType) {
+      case BackgroundMapType.Aerial:
+        return "Aerial Imagery";
+      default:
+      case BackgroundMapType.Hybrid:
+        return "Aerial Imagery with labels";
+      case BackgroundMapType.Street:
+        return "Streets";
+    }
+  }
+  private providerLabel() {
+    let label;
+    switch (this.providerName) {
+      case "BingProvider":
+      default:
+        label = `Bing Maps`;
+        break;
+
+      case "MapBoxProvider":
+        label = `MapBox`;
+        break;
+    }
+    return label;
+  }
+
+  public label() {
+    return  `${this.providerLabel()}: ${this.mapTypeName()}`;
+  }
+
+  public toMapSettings(): MapLayerSettings {
+    let formatId: string, url: string;
+    switch (this.providerName) {
+      case "BingProvider":
+      default:
+        formatId = "BingMaps";
+
+        let imagerySet;
+        switch (this.mapType) {
+          case BackgroundMapType.Street:
+            imagerySet = "Road";
+            break;
+          case BackgroundMapType.Aerial:
+            imagerySet = "Aerial";
+            break;
+          case BackgroundMapType.Hybrid:
+          default:
+            imagerySet = "AerialWithLabels";
+            break;
+        }
+        url = `https://dev.virtualearth.net/REST/v1/Imagery/Metadata/${imagerySet}?o=json&incl=ImageryProviders&key={bingKey}`;
+        break;
+
+      case "MapBoxProvider":
+        formatId = "MapboxImagery";
+        switch (this.mapType) {
+          case BackgroundMapType.Street:
+            url = "https://api.mapbox.com/v4/mapbox.streets/";
+            break;
+          case BackgroundMapType.Aerial:
+            url = "https://api.mapbox.com/v4/mapbox.satellite/";
+            break;
+          case BackgroundMapType.Hybrid:
+            url = "https://api.mapbox.com/v4/mapbox.streets-satellite/";
+            break;
+        }
+        break;
+    }
+    const name = this.label();
+    return MapLayerSettings.fromJSON({ name, formatId, url,
+      transparentBackground: false,
+      isBase: true,
+      visible: this.visible,
+      transparency: this.transparency })!;
+  }
+
+  public clone(changedProps: BackgroundMapProviderProps): BackgroundMapProvider {
+    if (undefined === changedProps)
+      return this;
+
+    const props = {
+      name: undefined !== changedProps.providerName ? changedProps.providerName : this.providerName,
+      providerData: {
+        mapType: undefined !== changedProps.providerData?.mapType ? changedProps.providerData.mapType : this.mapType,
+        visible: undefined !== changedProps.providerData?.visible ? changedProps.providerData.visible : this.visible,
+        transparency: undefined !== changedProps.providerData?.transparency ? changedProps.providerData.transparency : this.transparency,
```

Comment:
I'm unclear on why BackgroundMapProps no longer has a provider name/type but still has a transparency - is it redundant with this one?

Reply:
I agree its a bit confusing:
Transparency on BackgroundMapProps is an independent setting, it apply transparency on all components of mapImagery: baseMap, overlayLayers and BackgroundLayers.  Additionally, transparency can be set on independently foreach component (i.e MapLayerSettings.transparency, and now BackgroundMapProps.providerData.transparency).

Is there a business case for BackgroundMapProps.Transparency ?  I dont know, but its always been there and the map-layers widget expose it quite obviously.